src/uwtools/rocoto.py

```diff
@@ -181,14 +158,57 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param e: The parent element to add the new element to.
         :param config: Configuration data for this element.
         """
+        operands, operators, strequality = self._dependency_constants
         e = SubElement(e, STR.dependency)
-        for key, block in config.items():
-            tag, _ = self._tag_name(key)
-            if tag == STR.taskdep:
-                self._set_attrs(SubElement(e, STR.taskdep), block)
+
+        for tag, block in config.items():
+            tag, _ = self._tag_name(tag)
+            if tag in operands:
+                self._add_task_dependency_operand(e, block=block, tag=tag)
+            elif tag in operators:
+                self._add_task_dependency_operator(e, config={tag: block})
+            elif tag in strequality:
+                self._add_task_dependency_strequality(e, block=block, tag=tag)
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
+
+    def _add_task_dependency_operand(self, e: Element, block: dict, tag: str) -> None:
+        """
+        :param e: The parent element to add the new element to.
+        :param block: Configuration data for the tag.
+        :param tag: Configuration new element to add.
+        """
+        self._set_attrs(SubElement(e, tag), block)
+
+    def _add_task_dependency_operator(self, e: Element, config: dict) -> None:
+        """
+        Add an operator element to the <dependency>.
+
+        :param e: The parent element to add the new element to.
+        :param config: Configuration data for this element.
+        """
+
+        operands, operators, strequality = self._dependency_constants
+
+        for tag, block in config.items():
+            tag, _ = self._tag_name(tag)
+            if tag in operators:
+                self._add_task_dependency_operator(SubElement(e, tag), config=block)
+            elif tag in operands:
+                self._add_task_dependency_operand(e, tag=tag, block=block)
+            elif tag in strequality:
+                self._add_task_dependency_strequality(e, tag=tag, block=block)
             else:
                 raise UWConfigError("Unhandled dependency type %s" % tag)
 
+    def _add_task_dependency_strequality(self, e: Element, block: dict, tag: str) -> None:
+        """
+        :param e: The parent element to add the new element to.
+        :param block: Configuration data for the tag.
+        :param tag: Configuration new element to add.
+        """
+        self._set_attrs(SubElement(e, tag), block)
```

Comment:
It's interesting that this and the operand method both do exactly the same thing. Do we want to consider combining those?

Reply:
Pushing now that updates these changes.